go/api/base_client.go

```diff
@@ -3492,3 +3492,122 @@ func (client *baseClient) XClaimJustIdWithOptions(
 	}
 	return handleStringArrayResponse(result)
 }
+
+// Returns information about the stream stored at `key`.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	key - The key of the stream.
+//
+// Return value:
+//
+//	A stream information for the given `key`. See the example for a sample response.
+//
+// Example:
+//
+//	infoBreef, err := client.XInfoStream(key)
+//	infoBreef:
+//	// map[string]any {
+//	// 	"entries-added" : 1,
+//	// 	"first-entry" : []any{
+//	// 		"1719877599564-0", []any{"some_field", "some_value", ...},
+//	// 	},
+//	// 	"groups" : 1,
+//	// 	"last-entry" : []any{
+//	// 		"1719877599564-1", []any{"some_field", "some_value", ...},
+//	// 	},
+//	// 	"last-generated-id" : "1719877599564-1",
+//	// 	"length" : 1,
+//	// 	"max-deleted-entry-id" : "0-0",
+//	// 	"radix-tree-keys" : 1,
+//	// 	"radix-tree-nodes" : 2,
+//	// 	"recorded-first-entry-id" : "1719877599564-1",
+//	// }
+//
+// [valkey.io]: https://valkey.io/commands/xinfo-stream/
+func (client *baseClient) XInfoStream(key string) (map[string]any, error) {
```

Comment:
Is this able to delegate to `XInfoStreamWithOptions` by passing in `nil` for options.

Reply:
No, the command line and output is different. `XInfoStreamWithOptions` has always `FULL` argument.

Reply:
Is there ever a case where you would call WithOptions and then not provide an options object? It seems like calling WithOptions with a nil could be construed as a delegating call. If options is not null, then add "FULL" argument like you currently do.

Reply:
Done